### PR TITLE
feat(cards): #MAG-13(front) add CRUD for cards + display cards

### PR DESCRIPTION
**Pull request by Sofianebernoussi**
## Describe your changes
added CRUD for cards 
added displayed and modifiable cards

## Checklist tests
Check if we can get cards with a pagination
Check if we can update/delete cards
Check if the model is correctly instancied
Check if the lightbox are correctly configured

## Issue ticket number and link
https://jira.support-ent.fr/browse/MAG-13
https://jira.support-ent.fr/browse/MAG-34

- [X] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [X] If it is a consequent feature, I have added thorough tests.
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   },
   "homepage": "",
   "dependencies": {
-    "entcore": "4.5.1",
+    "entcore": "4.6.0-dev.0",
```
Comment (BastienBenard) [inline package.json:23]: Pourquoi ?

Comment (Sofianebernoussi) [reply package.json:23]: nouvelle maj de l'infra front

**Comment by BastienBenard** [comment]
Pense a mettre les doc d'API qui ont changé dans ta description

**Review by BastienBenard — APPROVED** [review]

File: src/main/resources/public/sass/global/components/_icons.scss [hunk 2]
```diff
@@ -79,6 +79,69 @@ i {
     content: '\F76F';
   }
 
+  &.magneto-card-settings::before {
+    font-family: 'material-icons';
```
Comment (JinDouang) [inline src/main/resources/public/sass/global/components/_icons.scss:83]: Attention au nommage material-icon @flmartineau

File: src/main/resources/public/sass/global/components/directives/_card-list-item-options.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
+  .card-list-item-options {
+    nav {
+      .nav, .sub-nav {
+        z-index: 1001;
+        position: absolute;
+        top: 0;
+        right: -113px;
+        z-index: 1;
+        background-color: #fff;
```
Comment (JinDouang) [inline src/main/resources/public/sass/global/components/directives/_card-list-item-options.scss:9]: variable

File: src/main/resources/public/ts/utils/date.utils.ts [hunk 4]
```diff
@@ -20,19 +22,51 @@ export class DateUtils {
...
         return moment(date).format(format);
     }
+
+    /**
+     * Format date based on given format using moment
+     * @param date date to format
+     * @param format format
+     */
+    static createdSince(date: any, format: string): string {
+        let now: any = moment();
```
Comment (JinDouang) [inline src/main/resources/public/ts/utils/date.utils.ts:49]: tu peux typer avec "Moment"

File: src/main/resources/public/ts/utils/date.utils.ts [hunk 5]
```diff
@@ -20,19 +22,51 @@ export class DateUtils {
...
+        let formatDate: any = moment(date);
+        let resultDate: string = "";
+        if (this.moreThan(formatDate, DAYS)) {
+            if (now.diff(formatDate, DAYS) == 1) {
+                resultDate = I18nUtils.translate("magneto.card.time.yesterday")
+            } else {
+                resultDate = formatDate.format(format);
+            }
+        } else if (this.moreThan(formatDate, HOURS)) {
+            resultDate = I18nUtils.getWithParams("magneto.card.time.hours", [now.diff(formatDate, HOURS)]);
```
Comment (JinDouang) [inline src/main/resources/public/ts/utils/date.utils.ts:59]: A revoir potentiellement la méthode "diff" te renvoie un number mais tu dis que c'est un string
(vu avec le typage Moment())

File: src/main/resources/public/ts/services/boards.service.ts [hunk 6]
```diff
@@ -34,6 +38,16 @@ export const boardsService: IBoardsService = {
             .then((res: AxiosResponse) => new Boards(res.data));
     },
 
+    getBoardsByIds: async (boardIds: Array<string>): Promise<Boards> => {
+        return http.post(`/magneto/boards`, {boardIds: boardIds})
+            .then((res: AxiosResponse) => new Boards(res.data));
+    },
+
+    getBoardById: async (boardId: string): Promise<Board> => {
+        return http.post(`/magneto/boards`, {boardIds: [boardId]})
```
Comment (JinDouang) [inline src/main/resources/public/ts/services/boards.service.ts:47]: C'est redondant avec la méthode au dessus

Tu peux très bien l'utiliser pour une seule collection dans ton controller/directive

File: src/main/resources/public/ts/services/boards.service.ts [hunk 7]
```diff
@@ -1,10 +1,14 @@
...
 import http, {AxiosResponse} from 'axios';
-import {IBoardsParamsRequest, Boards, BoardForm} from "../models";
+import {IBoardsParamsRequest, Boards, BoardForm, Board} from "../models";
 
 export interface IBoardsService {
     getAllBoards(params: IBoardsParamsRequest): Promise<Boards>;
 
+    getBoardsByIds(boardIds: Array<string>): Promise<Boards>;
+
+    getBoardById(boardId: string): Promise<Board>;
```
Comment (JinDouang) [inline src/main/resources/public/ts/services/boards.service.ts:10]: C'est redondant avec la méthode au dessus

File: src/main/resources/public/ts/directives/board-date/board-date.directive.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    constructor(private $scope: IBoardDateScope,
+                private $location: ILocationService,
+                private $window: IWindowService) {
+    }
+
+    $onInit() {
+    }
+
+    getDateText= (): string => {
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/board-date/board-date.directive.ts:31]: = ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Pull request merged**